src/sinks/splunk_hec/logs/tests.rs

```diff
@@ -197,3 +216,27 @@ async fn splunk_passthrough_token() {
         ]
     )
 }
+
+fn get_hec_data_for_timestamp_test(timestamp: Option<Value>, timestamp_key: &str) -> HecEventJson {
+    let processed_event = get_processed_event_timestamp(timestamp, timestamp_key);
+    let encoder = HecLogsEncoder::Json;
+    let bytes = encoder.encode_event(processed_event).unwrap();
+    serde_json::from_slice::<HecEventJson>(&bytes[..]).unwrap()
+}
```

Comment:
Nit: I might just define this function in the test function since it is only used there.